pkg/server/plugin/upstreamauthority/spire/spire_server_client.go

```diff
@@ -0,0 +1,132 @@
+package spireplugin
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"sync"
+
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"github.com/spiffe/spire/pkg/common/x509util"
+	"github.com/spiffe/spire/proto/spire/api/server/bundle/v1"
+	"github.com/spiffe/spire/proto/spire/api/server/svid/v1"
+	"github.com/spiffe/spire/proto/spire/common"
+	"github.com/spiffe/spire/proto/spire/types"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// newServerClient creates a new spire-sever client
+func newServerClient(serverAddr string, workloadapiSocket string) *serverClient {
+	return &serverClient{
+		serverAddr:        serverAddr,
+		workloadAPISocket: workloadapiSocket,
+	}
+}
+
+type serverClient struct {
```

Comment:
I'm not sure if serverClient is a good name. I'm wondering if there is a name that better describes the functionality that this struct provides.

Reply:
open to any suggestion